### ISSUE TITLE
Unable to open a file named "t_JNB01.ipynb"

### ISSUE THREAD

**Issue by djfm501** [microsoft/vscode#126788]
We have written the needed data into your clipboard because it was too large to send. Please paste.

Issue Type: <b>Bug</b>

Created a file named "t_JNB01.ipynb" (using another program eg touch).  Tried with a file size of 0 or more.
Tried to open it with vscode and get message "Unable to open 't_JBN01.ipynb': Unexpected string in JSON at position 2".  I assume that refers to the underscore in the filename.

I disabled all extensions and got the same problem.

I renamed the file to "t-JBN01.ipynb" and the file opened correctly.

I'm running Windows 10 and VSCode automatically updates.  VSCode version info:

Version: 1.57.1 (user setup)
Commit: 507ce72a4466fbb27b715c3722558bb15afa9f48
Date: 2021-06-17T13:28:07.755Z (3 days ago)
Electron: 12.0.7
Chrome: 89.0.4389.128
Node.js: 14.16.0
V8: 8.9.255.25-electron.0
OS: Windows_NT x64 10.0.19042

[OK] [Copy]

VS Code version: Code 1.57.1 (507ce72a4466fbb27b715c3722558bb15afa9f48, 2021-06-17T13:28:07.755Z)
OS version: Windows_NT x64 10.0.19042
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-3770K CPU @ 3.50GHz (8 x 3510)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|31.95GB (21.98GB free)|
|Process Argv|--crash-reporter-id fefeb82c-9607-4488-81f6-54ed4b4dcd5a|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (42)</summary>

Extension|Author (truncated)|Version
---|---|---
Bookmarks|ale|13.1.0
stackoverflow-search|Ale|1.2.0
vscode-file-templates|bam|1.1.0
bracket-pair-colorizer|Coe|1.0.61
vscode-eslint|dba|2.1.23
vscode-new-file|dku|4.0.2
python-preview|don|0.0.4
githistory|don|0.6.16
gitlens|eam|11.5.1
vscode-html-css|ecm|1.10.2
macros|ged|1.2.1
vscode-pull-request-github|Git|0.27.1
go|gol|0.26.0
vscode-back-forward-button|gri|0.1.5
macro-commander|jef|1.2.8
google-search|kam|0.0.1
MagicPython|mag|1.1.0
python|ms-|2021.6.944021595
vscode-pylance|ms-|2021.6.2
jupyter|ms-|2021.6.999406279
remote-wsl|ms-|0.56.5
hexeditor|ms-|1.7.1
powershell|ms-|2021.5.1
sqltools|mtx|0.23.0
sqltools-driver-sqlite|mtx|0.2.0
color-highlight|nau|2.3.0
back-n-forth|nic|3.1.0
autodocstring|njp|0.5.4
indent-rainbow|ode|7.5.0
coloured-comments|Par|0.5.0
git-file-history|pom|1.0.1
vscode-css-peek|pra|4.2.0
savebackup|pur|0.0.4
LiveServer|rit|5.6.1
better-pageup-pagedown|rus|0.0.1
markdown-preview-enhanced|shd|0.5.21
vscode-fileutils|sle|3.4.5
tab-groups|usa|1.4.8
errorlens|use|3.2.7
reopenclosedtab|uyi|1.1.0
vscode-jumpy|wma|0.3.1
markdown-all-in-one|yzh|3.4.0

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383cf:30185419
vspyt653:30270858
pythonvspyt602:30300191
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythonvspyt639:30300192
pythontb:30283811
pythonvspyt551cf:30311713
vspre833:30321513
pythonptprofiler:30281270
vshan820:30294714
pythondataviewer:30285071
vscus158:30321503
pythonvsuse255:30323308
vscorehov:30309549
vscod805:30301674
vscextlangct:30322307
binariesv615cf:30323120

```

</details>

**Comment by djfm501** [comment]
Correction.  I didn't try it with all extensions disabled because I didn't realize that you had to close VSCode and reopen it for the disable all extensions to take effect.
When I did restart VSCode "t_JNB01.ipynb" did open.

Further more, after re-enabling all extensions, restarting VSCode, the problems seems to have gone away.

**Comment by djfm501** [comment]
Well no the problem didn't go away.  I tried it with "t_JFN02.ipynb" and got the same problem.  Other than changing my naming conversion (I use 't_' in my .gitignore file) I don't have a solution.

**Comment by bpasero** [comment]
I wonder if this issue is being caused by an installed extension. Can you try to run VS Code without extensions? From the command line (NOT the integrated terminal in Code), execute: `code --disable-extensions` and try your steps again to see if it reproduces. If you see it is an issue with the extension, please file it against the extension repository itself.

**Comment by djfm501** [comment]
I did as suggested and ran "code --disable-extensions"  I was then able to open t_JNB01.ipynb without a problem.
I then closed code and reopened it normally.  Again I was able to open this file.
The problem went away.

Looking at {home}\.vscode\extensions\ I see that the following were updated on 2021-06-24 (yesterday):

- ms-toolsai.jupyter-2021.6.999662501
- ms-python.vscode-pylance-2021.6.3
- ms-vscode.powershell-2021.6.1

Maybe one of these updates fixed the problems.

Thanks for looking into this for me.
